model/build/build.go

```diff
@@ -166,6 +166,29 @@ func UpdateActivation(buildIds []string, active bool, caller string) error {
 	return errors.Wrapf(err, "setting build activation to %t", active)
 }
 
+// UpdateActivationAndStatus updates builds with the given ids
+// to the given activation setting and marks builds as created if active.
+func UpdateActivationAndStatus(buildIds []string, active bool, caller string) error {
+	query := bson.M{IdKey: bson.M{"$in": buildIds}}
+	if !active && evergreen.IsSystemActivator(caller) {
+		query[ActivatedByKey] = bson.M{"$in": evergreen.SystemActivators}
+	}
+	update := bson.M{
+		ActivatedKey:     active,
+		ActivatedTimeKey: time.Now(),
+		ActivatedByKey:   caller,
+	}
+	if active {
+		update[StatusKey] = evergreen.BuildCreated
```

Comment:
So, this isn't actually true in the case that we're only passing in _some_ buildIds (what if buildA is being activated but buildB is already running)? I recognize that isn't actually a concern since we're only using this function to update builds for an _entire_ version. That being said, should this function instead take in a versionId (rather than buildIds) and use that in the query, to indicate that this should only be used for full versions?